server.go

```diff
@@ -329,6 +332,73 @@ func (s *Server) Initialize(u *url.URL) error {
 	return nil
 }
 
+// Join creates a new data node in an existing cluster, copies the metastore,
+// and initializes the ID.
+func (s *Server) Join(u *url.URL, joinURL *url.URL) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	// Encode data node request.
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(&dataNodeJSON{URL: u.String()}); err != nil {
+		return err
+	}
+
+	// Send request.
+	joinURL = copyURL(joinURL)
+	joinURL.Path = "/data_nodes"
+	resp, err := http.Post(joinURL.String(), "application/octet-stream", &buf)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Check if created.
+	if resp.StatusCode != http.StatusCreated {
+		return ErrUnableToJoin
+	}
+
+	// Decode response.
+	var n dataNodeJSON
+	if err := json.NewDecoder(resp.Body).Decode(&n); err != nil {
+		return err
+	}
+	assert(n.ID > 0, "invalid join node id returned: %d", n.ID)
+
+	// Download the metastore from joining server.
+	joinURL.Path = "/metastore"
+	resp, err = http.Post(joinURL.String(), "application/octet-stream", &buf)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Update the ID on the metastore.
+	if err := s.meta.mustUpdate(func(tx *metatx) error {
+		return tx.setID(n.ID)
+	}); err != nil {
+		return err
+	}
+
+	// Set the ID on the server.
+	s.id = n.ID
+
+	return nil
+}
+
+// CopyMetastore writes the underlying metastore data file to a writer.
+func (s *Server) CopyMetastore(w io.Writer) error {
+	return s.meta.mustView(func(tx *metatx) error {
+		// Set content lengh if this is a HTTP connection.
+		if w, ok := w.(http.ResponseWriter); ok {
+			w.Header().Set("Content-Length", strconv.Itoa(int(tx.Size())))
+		}
+
+		// Write entire database to the writer.
+		return tx.Copy(w)
```

Comment:
`Tx.Copy()` is one of my favorite parts about Bolt. It makes a lot of operational stuff so much easier.